src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+//! Query network interface addresses
+//!
+//! Uses the Linux and/or BSD specific function getifaddrs to query the list
+//! of interfaces and their associated addresses.
+
+use std::ffi;
+use std::iter::Iterator;
+use std::mem;
+use std::option::Option;
+
+use libc;
+use {Result, Error};
+
+use sys::socket::InetAddr;
+
+#[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
```

Comment:
Oh, and: I used the module to reduce the amount of `#[cfg()]` conditions. I only used the module as I didn't find any good way to write this without repeating the conditions many times. I'm fine if you prefer the latter and changed things accordingly.